### ISSUE TITLE
matplotlibrc pip install bug

### ISSUE THREAD

**Issue by m3atwad** [matplotlib/matplotlib#20226]
### Bug report
I think this bug is due to a miss-match between matplotlib 3.4.2 and the matplotlibrc file on github

**Bug summary**
I am using pyinstaller to build a matplotlib application.  I notice that it fails to run due to the screenshot attached.  After investigating I've realized replacing the contents of my matplotlibrc file on my computer with the contents of the matplotlibrc file found here don't match:
[https://github.com/matplotlib/matplotlib/blob/v3.4.2/matplotlibrc.template](url)

the problem is fixed when I replace the contents of the file on my computer with the contents of the template on github.  I use pip to attempt to upgrade my matplotlib install, but it is 3.4.2 and pip3 says everything is up to date.  Oddly enough the matplotlibrc file on my computer doesn't match the one at that link.  It seems like i'm not pulling down the latest matplotlibrc file.  Why is this happening?

> **Matplotlib version**

  * Operating system:  Windows 10 64 bit
  * Matplotlib version 3.4.2
  * Matplotlib backend QT5Agg 
  * Python version: 3.7.9
  * Jupyter version (if applicable):  Not using
  * Other libraries: wx python for my gui, matplotlib, numpy

I installed matplotlib via pip3

Below is the contents of the matplotlibrc file from my pip3 install of matplot lib in my site-packages folder:

[local_matplotlibrc.txt](https://github.com/matplotlib/matplotlib/files/6476620/local_matplotlibrc.txt)
![46](https://user-images.githubusercontent.com/20314773/118223778-e0f02400-b436-11eb-883e-b118c3cb6f0e.jpg)

**Comment by tacaswell** [comment]
There is no screen shot attached so it is difficult to determine what is wrong.  I do not think pip compares the contents of the files under its management so it may not have noticed that you changed the file.

Given the issues you describe, and that you are using pyinstaller, I strongly suggest starting with a fresh virtual environment and installing everything from scratch, and then use pyinstaller to build your application.  There any many (many) ways for package installation to go wrong.  It is frequently faster to "turn it off and on again" and get a new environment than try to debug exactly _why_ the packaging has gone wrong!

**Comment by m3atwad** [comment]
Hi tacaswell - wow you replied fast.  I apologize for my poor post this was my first github post and I did a poor job.  

To simplify all of this - should the matplotlibrc file in my site-packages directory match the one below on github?
[](https://github.com/matplotlib/matplotlib/blob/v3.4.2/matplotlibrc.template)

**Comment by QuLogic** [comment]
It should be the same yes; you can download the wheel from PyPI, extract that file and compare them, and they are the same.

**Comment by m3atwad** [comment]
Hey Guys,

Thanks so much for the support especially since my post was quite poor.  I think I have some fundamental issues and the advice of using a virtual environment was really key.  I am not a software engineer or familiar with the virtual approach so did some tutorials and got up and running and just as you'd probably expect it was extremely helpful.  The problem I was running into, even though I don't understand what it was, is now solved with a clean venv setup with only the libraries I need and pyinstaller.  

So long story short this post should be closed and ultimately I needed to build a clean application in a much simpler environment which yielded a successful build.

Thanks again for the support and getting me going down the right path!
